src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,269 @@
+            feat_dynamic_cat=self.feat_dynamic_cat,
+            feat_static_real=self.feat_static_real,
+            feat_static_cat=self.feat_static_cat,
+            past_feat_dynamic_real=self.past_feat_dynamic_real,
+        )
+        if item_id:
+            dataentry["item_id"] = item_id
+        return dataentry
+
+    def __iter__(self) -> Iterator[DataEntry]:
+        for item_id, df in self._dataframes:
+            dataentry = self.process(self._dataentry(item_id, df))
+            if self.prediction_length:
+                dataentry = prepare_prediction_data(
+                    dataentry, self.prediction_length
+                )
+            yield dataentry
+
+    def __len__(self) -> int:
+        return len(self._dataframes)
+
+
+def LongDataFrameDataset(
+    dataframe: pd.DataFrame,
+    item_id: str,
+    target: Union[str, List[str]],
+    timestamp: str,
+    freq: str,
+    feat_dynamic_real: List[str] = [],
+    feat_dynamic_cat: List[str] = [],
+    feat_static_real: List[str] = [],
+    feat_static_cat: List[str] = [],
+    past_feat_dynamic_real: List[str] = [],
+    prediction_length: int = 0,
+) -> DataFramesDataset:
+    """
+    This dataset uses the long format for each variable. Target time series values,
+    for example, are stacked on top of each other rather than side-by-side. The
+    same is true for other dynamic or categorical features.
+
+    Parameters
+    ----------
+    dataframe: pd.DataFrame
+        pd.DataFrame containing at least timestamp, target and item_id columns.
+    target: str or List[str]
+        Name of the column that contains the target time series.
+        For multivariate targets `target` is expecting a list of column names.
+    timestamp: str
+        Name of the column that contains the timestamp information.
+    item_id: str
+        Name of the column that, when grouped by, gives the different time series.
+    freq: str
+        Frequency of observations in the time series. Must be a valid pandas frequency.
+    feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features.
+    feat_dynamic_cat: List[str]
+        List of column names that contain dynamic categorical features.
+    feat_static_real: List[str]
+        List of column names that contain static real features.
+    feat_static_cat: List[str]
+        List of column names that contain static categorical features.
+    past_feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features only for the history.
+    prediction_length: int
+        For target and past dynamic features last `prediction_length` elements are
+        removed when iterating over the data set. This becomes important when the
+        predictor is called.
+
+    Returns
+    -------
+    DataFramesDataset
+        A dataset based on a collection of pd.DataFrame-objects.
+    """
+    dataframes = dict(list(dataframe.groupby(item_id)))
```

Comment:
Nice!

Reply:
The `list` in there looks unnecessary.

Reply:
The casting to list is indeed needed. Otherwise there is a TypeError.
```python
import numpy as np; import pandas as pd
a = pd.DataFrame(np.random.normal(size=(5,2)))
a["x"] = 3*['a'] + 2*['b']
#dict(a.groupby('x')) # not working
dict(list(a.groupby('x'))) # working
```